packages/sst/src/constructs/StaticSite.ts

```diff
@@ -674,96 +660,65 @@ interface ImportMeta {
+        }),
+      ],
     });
-    this.bucket.grantReadWrite(uploader);
-    assets.forEach((asset) => asset.grantRead(uploader));
+    stack.customResourceHandler.role?.attachInlinePolicy(policy);
 
-    // Create the custom resource function
-    const handler = new Function(this, "S3Handler", {
-      code: Code.fromAsset(
-        path.join(__dirname, "../support/base-site-custom-resource")
-      ),
-      layers: [cliLayer],
-      runtime: Runtime.PYTHON_3_11,
-      handler: "s3-handler.handler",
-      timeout: Duration.minutes(15),
-      memorySize: 1024,
-      environment: {
-        UPLOADER_FUNCTION_NAME: uploader.functionName,
-      },
-    });
-    this.bucket.grantReadWrite(handler);
-    filenamesAsset?.grantRead(handler);
-    uploader.grantInvoke(handler);
-
-    // Create custom resource
-    return new CustomResource(this, "S3Deployment", {
-      serviceToken: handler.functionArn,
-      resourceType: "Custom::SSTBucketDeployment",
+    const resource = new CustomResource(this, "S3Uploader", {
+      serviceToken: stack.customResourceHandler.functionArn,
+      resourceType: "Custom::S3Uploader",
       properties: {
-        Sources: assets.map((asset) => ({
-          BucketName: asset.s3BucketName,
-          ObjectKey: asset.s3ObjectKey,
+        sources: assets.map((asset) => ({
+          bucketName: asset.s3BucketName,
+          objectKey: asset.s3ObjectKey,
         })),
-        DestinationBucketName: this.bucket.bucketName,
-        Filenames: filenamesAsset && {
-          BucketName: filenamesAsset.s3BucketName,
-          ObjectKey: filenamesAsset.s3ObjectKey,
+        destinationBucketName: this.bucket.bucketName,
+        filenames: filenamesAsset && {
+          bucketName: filenamesAsset.s3BucketName,
+          objectKey: filenamesAsset.s3ObjectKey,
         },
-        FileOptions: (fileOptions || []).map((o) => {
-          const { filters, cacheControl, contentType, contentEncoding } =
-            o as StaticSiteFileOptions;
-          const { include, exclude } = o as StaticSiteFileOptionsDeprecated;
-          return [
-            ...(typeof exclude === "string" ? [exclude] : exclude ?? [])
-              .map((entry) => ["--exclude", entry])
-              .flat(2),
-            ...(typeof include === "string" ? [include] : include ?? [])
-              .map((entry) => ["--include", entry])
-              .flat(2),
-            ...(filters || [])
-              .map((filter) =>
-                Object.entries(filter).map(([key, value]) => [
-                  `--${key}`,
-                  value,
-                ])
-              )
-              .flat(2),
-            cacheControl ? ["--cache-control", cacheControl] : [],
-            contentType ? ["--content-type", contentType] : [],
-            contentEncoding ? ["--content-encoding", contentEncoding] : [],
-          ].flat();
-        }),
-        ReplaceValues: this.getS3ContentReplaceValues(),
+        fileOptions: [
```

Comment:
The `fileOptions` are parsed first to last so the specificity of the `files` glob should be most specific to least specific (reverse of current implementation).

Reply:
Can be resolved by reversing file order.